frontend/src/app/Checkin/end.tsx

```diff
@@ -0,0 +1,120 @@
+import { StatusBar, StyleSheet, Text, View } from "react-native";
+import { useCallback } from "react";
+
+import Mascots from "@/assets/Poc_Mascots.svg";
+import Button from "@/components/Button";
+import { useAuth } from "@/contexts/userContext";
+import env from "@/util/validateEnv";
+
+import { lightModeColors } from "@/constants/colors";
+
+export default function Start() {
+  const { firebaseUser, refreshMongoUser } = useAuth();
+
+  const handleFinish = async () => {
+    try {
+      if (!firebaseUser) {
+        // No logged-in user; let the button still navigate home
+        return;
+      }
+
+      const idToken = await firebaseUser.getIdToken();
+
+      const res = await fetch(`${env.EXPO_PUBLIC_BACKEND_URI}/api/users/last-completed-weekly`, {
+        method: "PUT",
+        headers: {
+          Authorization: `Bearer ${idToken}`,
+          "Content-Type": "application/json",
+        },
+      });
+
+      if (!res.ok) {
+        console.warn("Failed to update weekly check-in timestamp", res.status);
+      } else {
+        // Refresh local mongo user so UI reflects updated timestamp
+        try {
+          await refreshMongoUser();
+        } catch (e) {
+          // non-fatal
+          console.warn("Failed to refresh mongo user after updating weekly check-in", e);
+        }
+      }
+    } catch (e) {
+      console.error("Error updating weekly check-in:", e);
+    }
+  };
+
+  return (
+    <View style={styles.container}>
+      <View style={styles.topSection}>
+        <Mascots style={styles.logo} />
+        <Text style={styles.title}>Thank You!</Text>
+        <Text style={styles.text}>
+          Lorem ipsum odor amet, consectetuer adipiscing elit. Curae phasellus laoreet nullam in
```

Comment:
Placeholder text again, can you replace this with a message about how checking in helps us ensure you're on the right track or smth